src/GameEntityTest.java

```diff
@@ -0,0 +1,193 @@
+import components.simplereader.SimpleReader;
+import components.simplereader.SimpleReader1L;
+import components.simplewriter.SimpleWriter;
+import components.simplewriter.SimpleWriter1L;
+
+/**
+ * Utility class with implementation of parts of Kernel methods in GmaeEntity.
+ *
+ * @author Joonie Kim
+ *
+ */
+public final class GameEntityTest {
+
+    /**
+     * Private constructor so this utility class cannot be instantiated.
+     */
+    private GameEntityTest() {
+
+    }
+
+    /**
+     * Moves the entity at map[y][x] to forward by its' direction.
+     *
+     * @param map
+     *            map field of the entity exist.
+     * @param x
+     *            x coordinate of the entity.
+     * @param y
+     *            y coordinate of the entity.
+     * @updates map
+     */
+    public static void moveForward(int[][] map, int x, int y) {
```

Comment:
Ideally, these would be instance methods. That said, looks good. Be careful with passing around aliases. This method should probably have an @updates parameter mode listed.